### PR TITLE
Shooter pid

### PR DESCRIPTION
**Pull request by ManiacMoustache**
Adds pid to shooterMotor and coast methods to motorHelper's sparkMax

### CODE REVIEW

File: utils/motorHelper.py [hunk 1]
```diff
@@ -214,13 +219,27 @@ def setupPid(self):
...
+        elif type == "Duty Cycle":
+            self.ControlType = rev.ControlType.kDutyCycle
+
+    def coast(self):
+        self.prevControlType = self.ControlType
+        self.setControlType("Duty Cycle")
+        self.set(0)
+
+    def stopCoast(self):
+        self.ControlType = self.prevControlType
```
Comment (tuffinmuffin) [inline utils/motorHelper.py]: If speed was never set to zero, code will crash here. Need to define the self.prevControLType prior to use.

File: components/loaderLogic.py [hunk 2]
```diff
@@ -19,6 +19,11 @@ class LoaderLogic(StateMachine):
     loaderMotorSpeed = tunable(.4)
     intakeMotorMinSpeed = tunable(.5)
     intakeMotorMaxSpeed = tunable(.7)
+    shootingSpeed = tunable(5300)
+    targetShootingSpeed = tunable(shootingSpeed - 300)
+
+    # Other variables
+    shooterStoppingDelay = 2
```
Comment (DrKittyKat) [inline components/loaderLogic.py]: This isn't used, we should delete it

File: components/shooterLogic.py [hunk 3]
```diff
@@ -13,12 +13,13 @@ class ShooterLogic(StateMachine):
...
     xboxMap: XboxMap
-
+    intShootingSpeed = 5300
     # Tunables
     loaderMotorSpeed = tunable(.4)
     intakeMotorMinSpeed = tunable(.5)
     intakeMotorMaxSpeed = tunable(.7)
-    targetShootingSpeed = tunable(5300)
+    shootingSpeed = tunable(intShootingSpeed)
+    targetShootingSpeed = tunable(intShootingSpeed - 50)
```
Comment (tuffinmuffin) [inline components/shooterLogic.py]: I think this should be a % of allowable error or a ram error window off the set point. this way there are 2 variables to juggle instep

File: .github/workflows/robot_ci.yml [hunk 4]
```diff
@@ -28,7 +28,7 @@ jobs:
         
     - name: Robotpy tests
       timeout-minutes: 1
-      continue-on-error: True
+      continue-on-speedTolerance: True
```
Comment (tuffinmuffin) [inline .github/workflows/robot_ci.yml]: global replace of "error" broke everything.

**Comment by tuffinmuffin** [comment]
@ChrisHirsch

File: .github/workflows/robot_ci.yml (hunk 4 above)
Comment (ManiacMoustache) [reply .github/workflows/robot_ci.yml]: Yeah, not the smartest on my part.

File: components/shooterLogic.py [hunk 5]
```diff
@@ -52,9 +54,10 @@ def doneShooting(self):
     @feedback
     def isShooterUpToSpeed(self):
         """Determines if the shooter is up to speed, then rumbles controller and publishes to NetworkTables."""
+        self.autoLoadingSpeed = self.shootingSpeed - self.speedTolerance
         if not self.isSetup:
             return False
-        atSpeed = bool(self.shooterMotors.shooterMotor.getEncoder().getVelocity() >= self.targetShootingSpeed)
+        atSpeed = bool(self.shooterMotors.shooterMotor.getEncoder().getVelocity() >= self.autoLoadingSpeed)
```
Comment (tuffinmuffin) [inline components/shooterLogic.py]: No need to cache "autoLoadingSpeed in object. Just calculate it here. If wanted keep it as a local. Remove from on_enable also. As this is a derived value caching it this way means any time either value is updated it has a potential to be an issue.

**Pull request merged**